### ISSUE TITLE
hover_data in PX doesn't accept df.index

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.py#2760]
In wide-mode I can't do this:

```python
import plotly.express as px
df = px.data.stocks()
fig = px.line(df, hover_data={df.index: "|%B %d, %Y"} )
```

or find any other way of referring to the index.

**Comment by nicolaskruchten** [comment]
So that works to add a *new* entry to the hoverlabel with label "index" but in my use-case is that in wide-form the index of `df` is used by default (in this case it's called `"date"`) and I can't format that hover entry:

![image](https://user-images.githubusercontent.com/203523/92759720-04ba1200-f35e-11ea-8c59-76a0d85ab42a.png)

**Comment by nicolaskruchten** [comment]
Ah but you're totally right, this *does* work:

```python
import plotly.express as px
df = px.data.stocks(indexed=True)
fig = px.line(df, hover_data={"date": ("|%Y", df.index)})
fig.show()
```

**Comment by nicolaskruchten** [comment]
Or more generally: `hover_data={df.index.name: ("|%Y", df.index)}` ... pretty obscure tho! Not sure where to document that one.

**Comment by imadcat** [comment]
> Or more generally: `hover_data={df.index.name: ("|%Y", df.index)}` ... pretty obscure tho! Not sure where to document that one.

this works for single index, what about multiple index?

**Comment by rabyj** [comment]
> Or more generally: `hover_data={df.index.name: ("|%Y", df.index)}` ... pretty obscure tho! Not sure where to document that one.

Yes the reference to the general syntax the first element of the tuple would be very appreciated. I finally found that the correct syntax for a string was `{"index": (df.index)}` but I had to fumble a lot.
